backend/src/halo2_impl.rs

```diff
@@ -11,6 +11,28 @@ impl<T: FieldElement> BackendImpl<T> for Halo2Prover {
         Halo2Prover::new(degree)
     }
 
+    fn add_verification_key(
+        &mut self,
+        pil: &Analyzed<T>,
+        fixed: &[(String, Vec<T>)],
```

Comment:
These two arguments probably should be given in the object's creation.

Reply:
I'm not sure. What if you still want to generate the verification key? For that you need more than just easy data like the size, field and backend type. You also need the optimized pil and fixed columns. The easiest way to get these is via Pipeline.

Reply:
I mean that they are used both in proving and verifying. Passing them on create, you can create the Halo2 `Plaf` once and store in the object, then it will be used for proving and verifying.